app/main.py

```diff
@@ -1,2 +1,67 @@
+from typing import Any, Hashable, List, Optional
+from fractions import Fraction
+
+
+class Node:
+    def __init__(self, key: Hashable, value: Any) -> None:
+        self.key = key
+        self.value = value
+        self.hash = hash(key)
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.size = 0
+        self.load_factor = Fraction(2, 3)
+        self.table: List[Optional[List[Node]]] = [None] * self.capacity
+
+    def _resize(self) -> None:
+        old_table = self.table
+        self.capacity *= 2
+        self.table = [None] * self.capacity
+        self.size = 0
+        for bucket in old_table:
+            if bucket:
+                for node in bucket:
+                    self.__setitem__(node.key, node.value)
+
+    def _hash_index(self, key: Hashable) -> int:
+        return hash(key) % self.capacity
+
+    def _find_node(
+        self,
+        bucket: Optional[List[Node]],
+        key: Hashable
+    ) -> Optional[Node]:
+        if bucket:
```

Comment:
this method should look like so:
- you iterate through hash table in a while loop
- for each element you check hash and key
- if it's correct then you return the Node and exit loop (or maybe finding index would be more correct so you can reuse this method for both get and set)
- if not found - continue going thorugh the table